applications/modbus_controller/targets/custom-cc3220sf/main.cxx

```diff
@@ -549,8 +549,11 @@ int appl_main(int argc, char *argv[])
     }
 
     resetblink(WIFI_BLINK_CONNECTING);
+    string mDNSServiceName = openlcb::TcpDefs::MDNS_SERVICE_NAME_GRIDCONNECT_CAN;
+    mDNSServiceName.append(".");
+    mDNSServiceName.append(openlcb::TcpDefs::MDNS_PROTOCOL_TCP);
     g_socket_client = new SocketClient(stack.service(),
-        openlcb::TcpDefs::MDNS_SERVICE_NAME_GRIDCONNECT_CAN, WIFI_HUB_HOSTNAME,
+        mDNSServiceName.c_str(), WIFI_HUB_HOSTNAME,
```

Comment:
this won't work because the lifetime of the new pointer you pass in is stack-lived whereas the SocketClient will probably hold on to the const char* passed in.
Maybe we could have two static symbols, one with the ._tcp suffix and another without? like basename and full name.
Any binary will only use one of the two.

Reply:
Yes, I'm thinking that makes sense, looking at that AvaHi docs again it looks like it expects the name to be in one string same as the SocketClient does. I'll add defines for both.